### PR TITLE
Create Sieve of Eratosthenes issue #308

### LINKED ISSUE #308: Java implementation of Sieve of Eratosthenes to generate prime numbers.
Please include a small description or summary of the issue here.

**Is your feature request related to a problem? Please describe.**
It will help people to understand the implementation Sieve of Eratosthenes to generate prime numbers in O(n*log(log(n))) time complexity.

**Describe the solution you'd like**
A standard implementation of Sieve of Eratosthenes.

**Languages**

-   [ ] Python
-   [x] Java
-   [ ] C
-   [ ] C++

### PR DESCRIPTION
**Pull request by prateekrai123**
Sieve of Eratosthenes implementation in Java

Please include a summary of the changes you have made in regard with this issue. 

Fixes #308 

## Type of change

**Please delete options that are not relevant**

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] This change requires a documentation update

# Checklist:

-   [x] I have read the contribution guidelines before making this PR.
-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented in my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation (Language folder or any other readme instructed to)
-   [x] My changes generates no new warnings
-   [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by prateekrai123** [comment]
In reference to issue #308

**Comment by prateekrai123** [comment]
@kaustubhgupta please review.

File: Java/sieveOfEratosthenes.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+public class sieveOfEratosthenes {
+
+	public static void main(String[] args) {
+		
+		Scanner sc = new Scanner(System.in);
+		int num = sc.nextInt();
+		//Creating a boolean array prime[0..size]
+		boolean prime[] = new boolean[num+1];
+		//Initializing each item in prime[] with true value
+		for(int i=0;i<=num;i++) {
```
Comment (YashSaxena9) [inline Java/sieveOfEratosthenes.java]: You can work without setting them true, and actually its preferred to use it without setting values to true as it saves extra efforts.

Comment (prateekrai123) [reply Java/sieveOfEratosthenes.java]: Sure.

**Review by kaustubhgupta — APPROVED** [review]

**Pull request merged**